src/client/client.ts

```diff
@@ -275,6 +281,65 @@ export class Client {
     return res;
   }
 
+  /**
+   * Batch publish messages to QStash.
+   */
+  public async batch(
+    req: PublishRequest[]
+  ): Promise<(PublishToUrlResponse | PublishToTopicResponse)[]> {
+    const messages = [];
+    for (const message of req) {
+      const headers = this.processHeaders(message);
+      const headerEntries: Record<string, string> = {};
+      for (const [key, value] of headers.entries()) {
+        headerEntries[key] = value;
+      }
+
+      messages.push({
+        destination: message.url ?? message.topic,
+        headers: headerEntries,
+        body: message.body,
+      });
+    }
+
+    const res = await this.http.request<
+      (PublishToUrlResponse | PublishToTopicResponse)[]
+    >({
+      path: ["v2", "batch"],
+      body: JSON.stringify(messages),
+      headers: {
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+    });
+
+    return res;
+  }
+
+  /**
+   * Batch publish messages to QStash, serializing each body to JSON.
+   */
+  public async batchJSON<
+    TBody = unknown,
+    TRequest extends PublishRequest<TBody> = PublishRequest<TBody>
+  >(
+    req: TRequest[]
+  ): Promise<(PublishToUrlResponse | PublishToTopicResponse)[]> {
+    for (const message of req) {
+      if ("body" in message) {
+        // @ts-ignore we can make the body a string from the JSON
+        message["body"] = JSON.stringify(message["body"]);
```

Comment:
Let's turn this into this `message.body`, otherwise biome linter complains. 

Or we can do this  `message.body = JSON.stringify(message.body) as TBody` since we know TBody is the desired format for body.

Reply:
Thank you!